File: typescript/packages/bench/src/rules/reworkLimitRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class ReworkLimitRule implements RuleContract {
    readonly key = "rework-limit";

    evaluate(data: Dataset): boolean {
        return data.orders.length >= 1;
    }
}

File: rust/crates/atelier-bench/src/rules/rework_limit_rule.rs
//! Rule 11/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct ReworkLimitRule;

impl RuleContract for ReworkLimitRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.orders.is_empty()
    }
}
